### ISSUE TITLE
Trying to contact you

### ISSUE THREAD

**Issue by johncarlson21** [TheThomasD/GeeetechA30T#1]
not sure exactly how to message you and you don't have discussions on this project, but I also have an A30T and I am very interested in how far you got with the marlin 2 update.

**Comment by moccor** [comment]
I have not yet, I honestly tried to look back at it recently but noticed how much of a pain github is to navigate and then forgot about it. I do consider it more and more, since PID autotune and estep calibration is either non-existent or requires connecting directly thru USB with their proprietary, hasn't been updated in 4 years, software. If I do try it out I will let you know. Atm I know nothing about flashing firmware to printers so I would have to be comfortable with that and going back to stock if needed.

**Comment by johncarlson21** [comment]
Sounds like just need to replace the board completely with another board setup for dual z and 3 extruders, like a creality board or something, and then get an Ender 3V2 screen :)

**Comment by johncarlson21** [comment]
Honestly I've had nothing but problems with this printer since I got it, base frame is tweaked, sd card slot doesn't read half the time, or reads and half way through print, can no longer read it etc among other things :|

**Comment by moccor** [comment]
Well I highly recommend Octoprint if you haven't used it. You almost never need the screen of a printer if you use Octoprint. The only time the screen is useful to me, is for bed leveling. And ofc you only need that one time unless something bad happens to the bed. Octoprint is basically a whole OS that gives wayyy more info and customization than that little A30T screen could ever give.

As for the problems, I am kinda 50/50. It is my first printer and I didn't know it was going to be such a pain and locked down/changed in some ways compared to an Ender (since the A30T literally is like a Ender clone). It's a extremely good deal for the size and performance, but I have been having problems with printing smaller things and having the supports peel away fine without breaking it. In comparison, my brothers Dremel 3D45 which cost 2000$ (4x more, for wayyyy smaller prints), prints those things fine, however the Dremel slicer likes to make things heavier than intended.

But if you forget about the screen and SD slot, and buy a Raspberry Pi 3/4 and install Octoprint, even a 100$ printer can feel like a 2000$ printer. It honestly surprises me how well the A30T performs with larger sized prints (def haven't used even half of he 320x420) but it works great. Even simple small things work great. I have been working on making a small 20 key keyboard using my A30T, and it surprised me how well the PCB and case printed out. But when I try to print small confusing mouse endoskeleton pieces, I am not as proud of it. But I feel like it is like others will say, its a very hard printer to get it working good.

I am far from the most knowledgeable, but I have been learning and trying things and have it in a decent spot rn. If you have any questions and I am confident in the info I will def help.

**Comment by TheThomasD** [comment]
Hi @johncarlson21 ,

whohoo! My first issue!

I don't like handing out my e-mail, but for now this issue might be a first place to exchange info.

My A30T was my third printer, I now have 5 (Anycubic Chiron, Artillery Sidewinder X1, this A30T, Tronxy X5SA-400 and Reality CR-30). Of all of them, the A30T is by far the most challenging IMHO. This is why I also wanted to get Marlin to work on it. In general, the printing is quite OK for me with the original firmware, however, the oozing of the nozzle is really bad.

Similar to @moccor I'm using Repetier Server as my main way of printing things. In combination with a touch screen it makes for a really nice printing factory :)

As for the Marlin part: I already got Marlin running (including the display). Most of the stuff works (I documented what I implemented and what not in some of the documents in the touchscreen folder of this repo), but of course, some improvements can be made. Info about how to flash Marlin is the readme in the base folder. If you look at my fork of the Marlin repo, you can see [this branch](https://github.com/TheThomasD/Marlin/tree/2.0.x-geeetech-a30t). This is my current development status. If you don't touch the screen to quickly (i.e. too many commands after another) many things will work. Be aware, though, that I enabled the BL-Touch permanently. If you don't have one, you'll have to change my screen software a little bit.

If you have some more detailed questions, just post them here and I'll try to answer them. You could maybe create a new issue for all of them so other can follow more easily.

Cheers
Thomas

**Comment by johncarlson21** [comment]
@moccor I at this point was gonna setup an old Mac to connect to it all the time but at the moment it is being a pile and I can't reload the OS on it. Was thinking of a raspberry pi but haven't gotten that far. Don't want to put anymore money into this thing if I don't have to. Also the sd card issue could be the extension and not the actual board. Gonna have to test that. 

@TheThomasD Hey bud. Nice to meet you and I'm gonna have to look at your stuff a bit more. Also I was thinking of setting up a telegram group maybe to have to contact about this stuff you are working on. What ya think? That way we don't have to put out our emails here 😜

**Comment by johncarlson21** [comment]
@TheThomasD I'm guessing you mean this is the way you flashed the board:

`I basically followed [this](https://github.com/MarlinFirmware/Configurations/blob/0414e840df34a8dbafce07131db4d6cd1ddc9c6a/config/examples/Tronxy/X5SA/HOWTO-INSTALL.md#backup-your-chitu-firmare-optional-but-strongly-recommended) guide to backup the firmware (as the chip is quite similar). I connected a switch to the boot0 pins on the board (they are clearly marked) that allows me to easily switch to the flashing mode (connected = flash enabled). Then, via the STM Cube Programmer, I was able to connect and download the backup via the standard USB port. I had to switch the whole printer on as it seems that the 5V from the USB are not used. There is another jumper on the board that maybe allows to use the 5V via USB, but switching the printer on and off is OK for me, so I never tried to find out. Be careful to not fry your board if you try, though.`

**Comment by TheThomasD** [comment]
@johncarlson21
Yes, that worked for me. Just be sure to make a backup :) And, well, all the stuff I tried worked for me, but no guarantee that it works for you as well ;)

**Comment by johncarlson21** [comment]
@TheThomasD haha thanks.. I am going to give this a try.. but it says I have to put a jumper on? wanna make sure I got that correct as I will have to open the case.

also I created a telegram group if you want to join and we can have people join and test and give info about your marlin conversion.
here is the link: [https://t.me/+qlIpbz8vWBljNTNh](url)
As soon as you join I'll make you an admin

**Comment by johncarlson21** [comment]
@moccor you are welcome to join also

**Comment by carebear380** [comment]
@TheThomasD  @johncarlson21  - I have an A30T with the same board as you indicated Thomas that I'd love to get switched over to Marlin.  I'm not very proficient on coding etc and only been 3D printing for about 1.5 yrs so would likely need a "Instructions for dumby's". I've been trying to get joined to telegram group but can't seem to get it o work.  Would anyone be able to help me?

**Comment by moccor** [comment]
The link to the Telegram group is right above. For some reason I can't copy it and I don't think I can create a different link, maybe only the group owner can create a new link. But ThomasD's guide is spot on with everything. Only thing I wish I known first was that Marlin has a feature that waits for the bed to heat up before doing calibration probes, which you can edit out in the source before compiling. 

I can compile you the stock source ThomasD created if you want, but you really should consider searching YouTube/Google on how to do it yourself since Marlin has a bunch of cool features you can add/remove, as well as it is important to set your own X/Y offsets depending on bed clips and whatnot, which you will need your own measurements (or just generously guess)
